langworld_db_data/featureprofiletools/update_profiles_inventories.py

```diff
@@ -0,0 +1,96 @@
+from pathlib import Path
+
+from langworld_db_data.constants.paths import FEATURE_PROFILES_DIR, INVENTORIES_DIR
+from langworld_db_data.filetools.csv_xls import read_dicts_from_csv, write_csv
+
+
+def rename_value_in_profiles_and_inventories(
+    value_to_rename_id: str,
+    new_value_name: str,
+    input_feature_profiles_dir=FEATURE_PROFILES_DIR,
+    output_feature_profiles_dir=FEATURE_PROFILES_DIR,
+    input_inventories_dir=INVENTORIES_DIR,
+    output_inventories_dir=INVENTORIES_DIR,
+) -> None:
+    """
+    Replaces all the instances of a given value name in profiles and features_listed_values on a given value.
+    Works with both singular and combined values.
+    """
+    files_list = list(input_feature_profiles_dir.glob("*.csv"))
+    for file in files_list:
+        print("Opening " + file.name)
+        data_to_write = process_one_feature_profile(
+            value_to_rename_id,
+            new_value_name,
+            file,
+        )
+        write_csv(
+            data_to_write, output_feature_profiles_dir / file.name, overwrite=True, delimiter=","
+        )
+        print("Successfully written into csv-file")
+    data_to_write = process_and_save_features_listed_values(
+        value_to_rename_id, new_value_name, input_inventories_dir / "features_listed_values.csv"
+    )
+    if not output_inventories_dir.exists():
+        output_inventories_dir.mkdir(parents=True, exist_ok=True)
+    write_csv(
+        data_to_write,
+        output_inventories_dir / "features_listed_values.csv",
+        overwrite=True,
+        delimiter=",",
+    )
+
+
+def process_one_feature_profile(
+    value_to_rename_id: str,
+    new_value_name: str,
+    filepath: Path,
+) -> list[dict[str, str]]:
+
+    number_of_replacements = 0
+    data_from_file = read_dicts_from_csv(filepath)
+    data_to_write = []
+    for line in data_from_file:
+        if value_to_rename_id in line["value_id"]:
+            line_to_write = line
```

Comment:
```suggestion
            line_to_write = line
```
this does **not** create a copy. You're still changing the original dictionary!

Reply:
The dictionary is now copied by means of the method .`copy()`